persist/fs/files.go

```diff
@@ -96,53 +101,93 @@ func TimeFromFileName(fname string) (time.Time, error) {
 	return time.Unix(0, latestTimeInNano), nil
 }
 
-// InfoFiles returns all the completed info files in the given shard
-// directory, sorted by their corresponding block start times.
-func InfoFiles(shardDir string) ([]string, error) {
+// readCheckpointFile reads the checkpoint file and returns the digest stored
+// in the checkpoint file. Note that buf has digestSize bytes to avoid gc and
+// repeated memory allocation.
+func readCheckpointFile(shardDir string, blockStart time.Time, buf []byte) (uint32, error) {
```

Comment:
Instead of passing in a `buf` here can we perhaps create a struct that keeps the `buf` and then make this a function on the struct so that users of this method don't have to explicitly pass in a buffer.  I'd rather not have a lot of "C like" methods peppered around the project.